roblox/client.py

```diff
@@ -550,3 +550,47 @@ def get_base_gamepass(self, gamepass_id: int) -> BaseGamePass:
         Returns: A BaseGamePass.
         """
         return BaseGamePass(client=self, gamepass_id=gamepass_id)
+
+    # Catalog
+    def get_catalog_items(self, catalog_item_array: List[TypedDict[catalog_id: int, catalog_item_type: Literal[1, 2]]]) -> List[CatalogItem]:
+        """
+        Gets a catalog item with the passed ID.
+
+        The catalog is also known as the Avatar Shop or the Marketplace.
+
+        Arguments:
+            catalog_id: A Roblox catalog item ID.
+            catalog_item_type: The type of item. 1 for an asset, and 2 for a bundle.
+
+        Returns:
+            A list of CatalogItem.
+        """
+        try:
+            catalog_item_response = await self._requests.post(
+                url=self._url_generator.get_url(
+                    "catalog", "v1/catalog/items/details"
+                ),
+                data={"data": catalog_item_array}
```

Comment:
You probably don't realize this, but `data=` is the parameter for urlencoded data. This might work because some Roblox endpoints do accept urlencoded data (haven't tested it) but you should make this `json=` instead.